R/SurveyFit.R

```diff
@@ -142,59 +150,237 @@ SurveyFit <- R6::R6Class(
+    #'   variable each row corresponds to.
+    summary = function(aggregated_estimates) {
+      if (!is.data.frame(aggregated_estimates)) {
+        stop("'aggregated_estimates' must be a data frame ",
+             "returned by the aggregate() method.", call. = FALSE)
+      }
+      lhs_var <- as.character(self$formula()[[2]])
+      if (ncol(aggregated_estimates) > 1) {
+        by_var <- colnames(aggregated_estimates)[1]
+        wtd_ests <- data.frame(
+          create_wtd_ests(self, lhs_var, by = by_var),
+          method = "wtd"
+        )
+        mrp_ests_tmp <- by(
+          aggregated_estimates[, "value"],
+          aggregated_estimates[, by_var],
+          FUN = function(x) c(mean = mean(x), sd = sd(x))
+        )
+        mrp_ests_tmp <- sapply(mrp_ests_tmp, function(x) x)
+        mrp_ests <- data.frame(
+          t(mrp_ests_tmp),
+          by_var = colnames(mrp_ests_tmp),
+          method = "mrp"
+        )
+        colnames(mrp_ests)[3] <- by_var
+        lhs_binary <- force_factor(self$map()$sample()$survey_data()[, lhs_var])
+        raw_ests_tmp <- by(
+          lhs_binary,
+          self$map()$mapped_sample_data()[, by_var],
+          FUN = function(x) c(mean = mean(x), sd = sqrt(mean(x) * (1 - mean(x))/length(x)))
+        )
+        raw_ests_tmp <- sapply(raw_ests_tmp, function(x) x)
+        raw_ests <- data.frame(
+          t(raw_ests_tmp),
+          by_var = colnames(raw_ests_tmp),
+          method = "raw"
+        )
+        colnames(raw_ests)[3] <- by_var
+      } else{
+        wtd_ests <- data.frame(create_wtd_ests(self, lhs_var), method = "wtd")
+        mrp_ests <- data.frame(
+          mean = mean(aggregated_estimates$value),
+          sd = sd(aggregated_estimates$value),
+          method = "mrp"
+        )
+        lhs_binary <- force_factor(self$map()$sample()$survey_data()[, lhs_var])
+        raw_ests <- data.frame(
+          mean = mean(lhs_binary),
+          sd = sqrt(mean(lhs_binary) * (1 - mean(lhs_binary))/length(lhs_binary)),
+          method = "raw"
+        )
+      }
+      out <- rbind(mrp_ests, raw_ests, wtd_ests)
+      rownames(out) <- NULL
+      out
+    },
+    #' @description Plot takes the aggregated MRP estimates and produces a quick visualization of total and sub-population estimates.
+    #' @param aggregated_estimates The data frame returned by `aggregate`.
+    #' @param additional_stats A vector that specifies which of three additional
+    #'   stats ("wtd", "raw", "mrp", "none") should be included on the plot. The
+    #'   default value is to include the weighted estimate (wtd) and raw data
+    #'   mean (raw), but an analogous bar for MRP (mrp) can be added using the
+    #'   posterior mean and sd. The sd for the weighted estimate uses the survey
+    #'   design and the \pkg{survey} package, whilst the raw estimate is a
+    #'   direct mean and binomial sd of the binary responses. Intervals are by
+    #'   default 95% CI.
```

Comment:
I think a few things could be clarified in this doc: 

1. It would be good to mention what plots are made, i.e. either a violin plot or a density plot depending on whether the aggregated estimates are by group or for the population

2. That the additional stats are added as interval estimates when plotting sub-populations but just the mean when plotting the population (is that right?) 

3. It says the default is 95% CI but doesn't provide a way to change the default. If we're not going to allow changing it then I think it would be clear to just say that the interval is a 95% CI without referring to it as the "default" (default makes it sound like it's changeable by the user).